pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +91,186 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
 	}
+
+	errs, warns = validateHubDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	if warn := validateHubChannels(bundle.Channels); warn != nil {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
 	return result
 }
 
-func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+// validateHubChannels will check the channels. The motivation for the following check is to ensure that operators
+// authors knows if their operator bundles are or not respecting the Naming Convention Rules.
+// However, the operator authors still able to choose the names as please them.
+func validateHubChannels(channels []string) error {
+	const preview = "preview"
+	const stable = "stable"
+	const fast = "fast"
+	for _, channel := range channels {
+		if !strings.Contains(channel, preview) &&
+			!strings.Contains(channel, stable) &&
+			!strings.Contains(channel, fast) {
+			return fmt.Errorf("ensure that your channels are following the channel naming convention. " +
+				"More info: https://olm.operatorframework.io/docs/best-practices/channel-naming/")
+		}
+
+	}
+	return nil
+}
+
+// validateHubDeprecatedAPIS will check if the operator bundle is using a deprecated or no longer supported k8s api
+func validateHubDeprecatedAPIS(bundle *manifests.Bundle, k8sVersion string) ([]error, []error) {
 	var errs []error
 	var warns []error
+	var k8sVersionFound string
+	k8s1_22, _ := semver.Make("1.22.0")
+	k8s1_16, _ := semver.Make("1.16.0")
 
-	if csv.Spec.Provider.Name == "" {
-		errs = append(errs, fmt.Errorf("csv.Spec.Provider.Name not specified"))
+	// If the K8S version was informed via an flag/option then, we should use it
+	k8sVersionFound = k8sVersion
+	if len(strings.Split(k8sVersionFound, ".")) == 2 {
+		// if the value found is like 1.16 then, we concat with .0 to check it with semver
+		k8sVersionFound += ".0"
 	}
+	semK8s, _ := semver.Make(k8sVersionFound)
 
-	for _, maintainer := range csv.Spec.Maintainers {
-		if maintainer.Name == "" || maintainer.Email == "" {
-			errs = append(errs, fmt.Errorf("csv.Spec.Maintainers elements should contain both name and email"))
+	// Transform the spec minKubeVersion in semver Version to compare
+	if len(strings.Split(bundle.CSV.Spec.MinKubeVersion, ".")) == 2 {
+		bundle.CSV.Spec.MinKubeVersion += ".0"
+	}
+	semverMinKube, _ := semver.Make(bundle.CSV.Spec.MinKubeVersion)
+
+	// Check if the versions found/informed are that ones which the deprecated APIS are no longer supported (>= 1.22)
+	isUnsupported := semK8s.GE(k8s1_22) || semverMinKube.GE(k8s1_22)
+
+	// If the operator bundle is configured to support all k8s versions or any version >= 16
+	if len(k8sVersionFound) == 0 || semK8s.GE(k8s1_16) {
```

Comment:
Isn't this also true if you just don't pass a value in? Are we making the k8sVersion parameter required?

Reply:
I improved the code here to make that more clear + add some comments. 
- If the option key is informed than we use it
- (So, if that was not informed or it is >= 1.16 we need to do the check)

If the key is not informed we will consider the minKubeVersion. However, if the minKubeVersion is empty for example it means that the operator bundle can be distributed for any k8s version and we should check and raise a warning.  If the minKubeVersion >= 1.22 then, we can safely raise the issue instead.

Reply:
Closing that as solved. However, feel free to re-open if you see that is required.